relay/relay.go

```diff
@@ -496,13 +541,15 @@ func (r *Relay) handleEvents(ctx context.Context, reader2 reader.Reader, transfo
 
 		if _, ok := e.Event.(*replication.RotateEvent); ok && utils.IsFakeRotateEvent(e.Header) {
 			isNew, err2 := isNewServer(ctx, r.meta.UUID(), r.db.DB, r.cfg.Flavor)
+			// should start from the transaction beginning when switch to a new server
 			if err2 != nil {
-				return err2
+				return 0, err2
```

Comment:
maybe also return `eventIndex` so we could skip the events?

Reply:
No. We need to ensure the new file start with a complete transaction because we need to include the `PreviousGTIDSET` event at the file beginning.